app/gui2/src/stores/graph/index.ts

```diff
@@ -307,11 +307,23 @@ export const useGraphStore = defineStore('graph', () => {
     )
   }
 
-  function setNodeContent(id: NodeId, content: string) {
+  function setNodeContent(id: NodeId, content: string, withImports?: RequiredImport[] | undefined) {
     const node = db.nodeIdToNode.get(id)
     if (!node) return
     edit((edit) => {
       edit.getVersion(node.rootSpan).syncToCode(content)
+      if (withImports) {
+        const conflicts = addMissingImports(edit, withImports)
+        if (conflicts == null) return
+        const wholeAssignment = edit.getVersion(node.rootSpan)?.mutableParent()
```

Comment:
The `rootSpan` is not "wholeAssignment", only the right hand of it.

Reply:
We’re getting `mutableParent` here, so the naming is actually correct. The weirdness of the current `updateValue` API is that it can’t replace the root of the current expression, it can only replace some of the children.